src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,156 @@
+    }
+
+    @Test
+    public void register_nullAge_notOk() {
+        User user = createUser(VALID_LOGIN, VALID_PASSWORD, null);
+        assertThrows(RegistrationException.class,
+                () -> registrationServiceImpl.register(user));
+    }
+
+    @Test
+    public void register_tooLongLogin_notOk() {
+        User user = createUser(TOO_LONG_LOGIN, VALID_PASSWORD, VALID_AGE);
+        assertThrows(RegistrationException.class,
+                () -> registrationServiceImpl.register(user));
+    }
+
+    @Test
+    public void register_existingLogin_notOk() {
+        User user = createUser(EXISTING_LOGIN, VALID_PASSWORD, VALID_AGE);
+        storageDao.add(user);
+        assertThrows(RegistrationException.class,
+                () -> registrationServiceImpl.register(user));
+    }
+
+    @Test
+    public void register_invalidLogin_notOk() {
+        User user = createUser(INVALID_LOGIN, INVALID_PASSWORD, VALID_AGE);
+        assertThrows(RegistrationException.class,
+                () -> registrationServiceImpl.register(user));
+    }
+
+    @Test
+    public void register_validLogin_ok() throws RegistrationException {
+        User user = createUser(VALID_LOGIN, VALID_PASSWORD, VALID_AGE);
+        registrationServiceImpl.register(user);
+        assertEquals(user, storageDao.get(VALID_LOGIN));
+    }
+
+    @Test
+    public void register_tooLongPassword_notOk() {
+        User user = createUser(VALID_LOGIN, TOO_LONG_PASSWORD, VALID_AGE);
+        assertThrows(RegistrationException.class,
+                () -> registrationServiceImpl.register(user));
+    }
+
+    @Test
+    public void register_invalidPassword_notOk() {
+        User user = createUser(VALID_LOGIN, INVALID_PASSWORD, VALID_AGE);
+        assertThrows(RegistrationException.class,
+                () -> registrationServiceImpl.register(user));
+    }
+
+    @Test
+    public void register_validPassword_ok() throws RegistrationException {
+        User user = createUser(VALID_LOGIN, VALID_PASSWORD, VALID_AGE);
+        registrationServiceImpl.register(user);
+        assertEquals(user, storageDao.get(VALID_LOGIN));
+    }
+
+    @Test
+    public void register_ageGraeaterThanMax_notOk() {
+        User user = createUser(VALID_LOGIN, VALID_PASSWORD, GREATER_THAN_MAX_AGE);
+        assertThrows(RegistrationException.class,
+                () -> registrationServiceImpl.register(user));
+    }
+
+    @Test
+    public void register_invalidAge_notOk() {
+        User user = createUser(VALID_LOGIN, VALID_PASSWORD, INVALID_AGE);
+        assertThrows(RegistrationException.class,
+                () -> registrationServiceImpl.register(user));
+    }
+
+    @Test
+    public void register_validAge_ok() throws RegistrationException {
+        User user = createUser(VALID_LOGIN, VALID_PASSWORD, VALID_AGE);
+        registrationServiceImpl.register(user);
+        assertEquals(user, storageDao.get(VALID_LOGIN));
+    }
```

Comment:
only _ok test is needed, assert what you need there - that the returned object is as expected, and that newly added user can be retrieved from DB